renku/service/errors.py

```diff
@@ -16,7 +16,328 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """Renku service exceptions."""
+import os
+
 from renku.core.errors import RenkuException
+from renku.service.config import (
+    DOCS_URL_BASE,
+    DOCS_URL_ERRORS,
+    ERROR_NOT_AVAILABLE,
+    SVC_ERROR_PROGRAMMING,
+    SVC_ERROR_USER,
+)
+from renku.service.serializers.headers import OptionalIdentityHeaders
+
+HAS_SENTRY = True if os.getenv("SENTRY_DSN") else False
+SENTRY_URL = os.getenv("SENTRY_URL", None)
+
+
+class ServiceError(Exception):
+    """Error handler for service errors.
+
+    Set the documentation reference for developers willing to explore the base code.
+    Handle Sentry, adding user data (when available) and error details.
+    """
+
+    devReference = f"{DOCS_URL_BASE}{DOCS_URL_ERRORS}" + "{class_name}"
```

Comment:
Actually, the `"{class_name}"` is _not_ an f-string on purpose here. It's used only to provide meaningful autogenerated documentation since `self.__class__.__name__` doesn't work there.
The proper class name is set in the `__init__` function.

![Screenshot_20220210_163910](https://user-images.githubusercontent.com/43481553/153442184-38d78cab-7ba3-4d58-b484-19a357637944.png)

Reply:
Ah right, I missed that